proto/structured.proto

```diff
@@ -27,61 +27,51 @@ option (gogoproto.sizer_all) = true;
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
-message Table {
-  optional string name = 1 [(gogoproto.nullable) = false];
+// For Cockroach internal use.
+message Internal {
+  optional uint32 id = 1 [(gogoproto.nullable) = false];
 }
-
+ 
 message Column {
   enum ColumnType {
     BYTES = 0;
   }
-
   optional string name = 1 [(gogoproto.nullable) = false];
   optional ColumnType type = 2 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
 }
 
 message Index {
   optional string name = 1 [(gogoproto.nullable) = false];
   optional bool unique = 2 [(gogoproto.nullable) = false];
+  // An ordered list of column names of which the index is comprised. Each
+  // column_name refers to a column in the TableSchema's columns. Special
+  // care is taken to update this when deleting columns.
+  repeated string column_names = 3 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
```

Comment:
Hey you know what I'll just online it. It's not a big deal really since we
are anyway going to continue iterating on this. I'm just happy we are in
lining this rather using the existing scheme.

On Sat, Jun 6, 2015, 3:10 PM Peter Mattis <notifications@github.com> wrote:

> In proto/structured.proto
> <https://github.com/cockroachdb/cockroach/pull/1278#discussion_r31870628>:
>
> >  }
> >
> >  message Index {
> >    optional string name = 1 [(gogoproto.nullable) = false];
> >    optional bool unique = 2 [(gogoproto.nullable) = false];
> > +  // An ordered list of column names of which the index is comprised. Each
> > +  // column_name refers to a column in the TableSchema's columns. Special
> > +  // care is taken to update this when deleting columns.
> > +  repeated string column_names = 3 [(gogoproto.nullable) = false];
> > +  // For Cockroach internal use.
> > +  optional Internal internal = 9 [(gogoproto.nullable) = true];
>
> Ok, guess I'm not clear on why you want to segregate the internal data vs
> inlining it in the structure that needs it. Right now the internal data is
> identical between table/column/index, but I doubt it will remain that way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1278/files#r31870628>.
>